docs/tutorials/time-series.md

```diff
@@ -0,0 +1,738 @@
+|-------------------------------------|----------------------------|------------------------------|------------------------------|-------------------------------------------------|-----------------------------|
+| `VALUES('2020-01-01 01:00:00', 10)` | '2020-01-01 01:00:00'      | '2020-01-01 00:00:00'        | '2020-01-01 00:00:00'        | `insert: {"d":"2020-01-01 00:00:00","total":10}`|                             |
+| `VALUES('2020-01-01 02:00:00', 10)` | '2020-01-01 02:00:00'      | '2020-01-01 01:00:00'        | '2020-01-01 00:00:00'        | `delete: {"d":"2020-01-01 00:00:00","total":10}`|                             |
+|                                     |                            |                              |                              | `insert: {"d":"2020-01-01 00:00:00","total":20}`|                             |
+| `VALUES('2020-01-02 00:00:00', 10)` | '2020-01-02 00:00:00'      | '2020-01-01 23:00:00'        | '2020-01-01 00:00:00'        | `insert: {"d":"2020-01-02 00:00:00","total":10}`|                             |
+| `VALUES('2020-01-02 01:00:00', 10)` | '2020-01-02 01:00:00'      | '2020-01-02 00:00:00'        | '2020-01-02 00:00:00'        | `delete: {"d":"2020-01-02 00:00:00","total":10}`| `insert: {"d":"2020-01-01 00:00:00","total":20}`|
+|                                     |                            |                              |                              | `insert: {"d":"2020-01-02 00:00:00","total":20}`|                             |
+
+Explanation:
+
+* The `purchase.ts` waterline trails one hour behind stream time due
+  to the `LATENESS` annotation on the `purchase` table.
+
+* The `daily_total.d` waterline is obtained by rounding the `purchase.ts`
+  waterline down to start of day using `TIMESTAMP_TRUNC()`.
+
+* Without `emit_final`, every input update produces an output update,
+  deleting any outdated records and inserting new records instead.
+
+* With `emit_final` only outputs below the current waterline are
+  produced. In this example, the waterline of the `daily_total.d`
+  timestamp column remains at '2020-01-01 00:00:00' until the last
+  `INSERT`, which moves the stream time to '2020-01-02 01:00:00'.
+  At this point, we know that no new inputs will have a timestamp
+  `< 2020-01-02 00:00:00` and can output the final aggregate value
+  for `2020-01-01`.
+
+The `emit_final` property must specify either a column name that
+exists in the view, or a column number, where 0 is the leftmost view
+column.
+
+The use of `emit_final` is subject to the following restrictions:
+
+* If the query engine cannot infer a waterline for the specified
+  column of the view, it will emit an error at compilation time.
+
+* Currently this annotation is only allowed on views that are not
+  [`LOCAL`](/sql/grammar/#creating-views).  It takes effect only
+  for the view used as output.  If the view is used in defining
+  other views, these derived views will receive the non-delayed data.
```

Comment:
I really liked my example about the one year old records, I think it's useful to understand this

Reply:
Added this

Note that using `emit_final` can significantly delay the output of a view.
In the example above, the aggregation is performed over a 1-day window,
resulting in a 1-day delay before the output is produced. If the aggregation
window were extended to 1 year, the view would not produce any outputs for
an entire year.